lib/ansible/template/__init__.py

```diff
@@ -399,9 +383,32 @@ def generate_ansible_template_vars(path: str, fullpath: str | None = None, dest_
         template_run_date=_datetime.datetime.now(),
         template_destpath=dest_path,
         template_fullpath=fullpath,
-        ansible_managed=ansible_managed,
     )
 
+    if include_ansible_managed:  # only inject the config default value if the variable wasn't set
+        managed_default = _constants.config.get_config_value('DEFAULT_MANAGED_STR')
+
+        managed_str = managed_default.format(
+            # IMPORTANT: These values must be constant strings to avoid template injection.
+            #            Use Jinja template expressions where variables are needed.
+            host="{{ template_host }}",
+            uid="{{ template_uid }}",
+            file="{{ template_path }}",
+        )
+
+        ansible_managed = _time.strftime(managed_str, _time.localtime(template_stat.st_mtime))
+        # DTFIX-RELEASE: this should not be tag_copy, it should either be an origin copy or some kind of derived origin
+        ansible_managed = _datatag.AnsibleTagHelper.tag_copy(managed_default, ansible_managed)
+        ansible_managed = trust_as_template(ansible_managed)
+        ansible_managed = _module_utils_datatag.deprecate_value(
+            value=ansible_managed,
+            msg="The `ansible_managed` variable is deprecated.",
+            help_text="Define and use a custom variable instead.",
```

Comment:
I think the deprecation message makes it sounds like ansible_managed as a variable is off-limits, even though defining it will suppress the deprecation.